src/key.rs

```
@@ -0,0 +1,137 @@
+        where
+            E: de::Error,
+        {
+            Ok(get_key_from_string(v))
+        }
+    }
+
+    return deserializer.deserialize_str(KeyVisitor);
+}
+
+#[derive(Debug)]
+pub struct Key {
+    pub code: KeyCode,
+    pub modifier: KeyModifiers,
+}
+
+impl Default for Key {
+    fn default() -> Self {
+        Self {
+            code: KeyCode::Null,
+            modifier: KeyModifiers::NONE,
+        }
+    }
+}
+
+fn get_key_code_from_string(string: &str) -> KeyCode {
+    if string.len() == 1 {
+        let key_code_char = string.chars().next();
+        if let Some(key_code_char) = key_code_char {
+            if key_code_char.is_lowercase() {
+                return KeyCode::Char(key_code_char);
+            }
+        }
+    }
+    match string {
+        "Backspace" => KeyCode::Backspace,
+        "Enter" => KeyCode::Enter,
+        "Left" => KeyCode::Left,
+        "Right" => KeyCode::Right,
+        "Up" => KeyCode::Up,
+        "Down" => KeyCode::Down,
+        "Home" => KeyCode::Home,
+        "End" => KeyCode::End,
+        "PageUp" => KeyCode::PageUp,
+        "PageDown" => KeyCode::PageDown,
+        "Tab" => KeyCode::Tab,
+        "BackTab" => KeyCode::BackTab,
+        "Delete" => KeyCode::Delete,
+        "Insert" => KeyCode::Insert,
+        "Esc" => KeyCode::Esc,
+        "CapsLock" => KeyCode::CapsLock,
+        "ScrollLock" => KeyCode::ScrollLock,
+        "NumLock" => KeyCode::NumLock,
+        "PrintScreen" => KeyCode::PrintScreen,
+        "Pause" => KeyCode::Pause,
+        "Menu" => KeyCode::Menu,
+        "KeypadBegin" => KeyCode::KeypadBegin,
+        _ => KeyCode::Null,
+    }
+}
+
+fn get_key_modifier_from_string(string: &str) -> KeyModifiers {
+    match string {
+        "C" => KeyModifiers::CONTROL,
+        "A" => KeyModifiers::ALT,
+        "W" => KeyModifiers::SUPER,
+        "H" => KeyModifiers::HYPER,
+        "M" => KeyModifiers::META,
+        _ => KeyModifiers::NONE,
```

An unknown modifier should probably be an error, not a silent fallback to no modifier.